connectors/byoei.py

```diff
@@ -130,11 +138,56 @@ async def _batch_bulk(self, operations):
                         if "error" in data:
                             logger.error(f"operation {op} failed, {data['error']}")
                             raise Exception(data["error"]["reason"])
+
+            self._populate_stats(operations, res)
+
         finally:
             self.bulk_time += time.time() - start
 
         return res
 
+    def _populate_stats(self, operations, res):
+        operation = doc_id = None
+        # A dictionary of operation -> an object, which is another dictionary of _id -> document size
+        ops_dict = {OP_INDEX: {}, OP_UPSERT: {}, OP_DELETE: {}}
+        for op in operations:
+            if OP_INDEX in op:
+                operation = OP_INDEX
+                doc_id = op[OP_INDEX]["_id"]
+                ops_dict[OP_INDEX][doc_id] = 0
+            elif OP_UPSERT in op:
+                operation = OP_UPSERT
+                doc_id = op[OP_UPSERT]["_id"]
+                ops_dict[OP_UPSERT][doc_id] = 0
+            elif OP_DELETE in op:
+                operation = OP_DELETE
+                doc_id = op[OP_DELETE]["_id"]
+                ops_dict[OP_DELETE][doc_id] = 0
+            elif "doc" in op:  # update payload
+                ops_dict[operation][doc_id] = get_byte_size(  # pyright: ignore
+                    op["doc"]
+                )
+            else:  # index payload
+                ops_dict[operation][doc_id] = get_byte_size(op)  # pyright: ignore
```

Comment:
I remember @tarekziade's concerns about `get_byte_size` performance degradation.

What was the outcome of this discussion?

Reply:
The history is here: https://github.com/elastic/connectors-python/pull/528#discussion_r1113432178

cc @tarekziade